pkgs/applications/misc/koreader/default.nix

```diff
@@ -20,9 +20,12 @@ stdenv.mkDerivation rec {
   src = if stdenv.isAarch64 then fetchurl {
     url = "https://github.com/koreader/koreader/releases/download/v${version}/koreader-${version}-arm64.deb";
     hash = "sha256-FwwB9slKOiYQ3eud2tiqov6yGNxmIicIe6nFpsH28Vk=";
-  } else fetchurl {
+  } else if stdenv.isx86_64 then fetchurl {
     url = "https://github.com/koreader/koreader/releases/download/v${version}/koreader-${version}-amd64.deb";
     hash = "sha256-hqJRZDZqzPNLK/8Bb+Oay70JqKAMKB0Epbbzeu5npLw=";
+  } else fetchurl {
+    url = "https://github.com/koreader/koreader/releases/download/v${version}/koreader-${version}-armhf.deb";
+    hash = "sha256-LgeWQcHm5Qq/7MUuidjily0WsOFZAWGWeO52jNHWKMw=";
   };
```

Comment:
Can you change this in to a thing like this
https://github.com/NixOS/nixpkgs/blob/8ce34b38ecba5be44fe665c6e779beadc1eae2d3/pkgs/by-name/ut/ut1999/package.nix#L53